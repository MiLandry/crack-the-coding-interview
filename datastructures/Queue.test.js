const Queue = require('./Queue');

let queue

beforeEach(() => {

queue = new Queue()
});

test('it adds', () => {
  queue.enqueue('a')
  queue.enqueue('b')
  queue.enqueue('c')
  queue.enqueue('d')
  expect(queue)
    .toMatchSnapshot()

})

test('it pops', () => {
  queue.enqueue('a')
  queue.enqueue('b')
  queue.enqueue('c')
  queue.enqueue('d')
  const personInline = queue.dequeue()
  expect(personInline)
    .toEqual('a')
  expect(queue)
    .toMatchSnapshot()

})
