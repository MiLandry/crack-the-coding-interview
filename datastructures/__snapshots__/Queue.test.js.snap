// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it adds 1`] = `
Queue {
  "data": Array [
    "a",
    "b",
    "c",
    "d",
  ],
}
`;

exports[`it pops 1`] = `
Queue {
  "data": Array [
    "b",
    "c",
    "d",
  ],
}
`;
